resources/prometheus/unit_tests/RHACSEgressProxy.yaml

```diff
@@ -0,0 +1,52 @@
+rule_files:
+  - /tmp/prometheus-rules-test.yaml
+
+evaluation_interval: 1m
+
+tests:
+  - interval: 1m
+    input_series:
+      - series: kube_deployment_status_replicas_ready{namespace="rhacs-aaaabbbbccccddddeeee", deployment="egress-proxy", container="kube-rbac-proxy-main"}
+        values: "3x10 2x10 1x20"
+    alert_rule_test:
+      - eval_time: 10m
+        alertname: RHACSEgressProxyReplicaCount
+        exp_alerts: []
+      - eval_time: 20m
+        alertname: RHACSEgressProxyReplicaCount
+        exp_alerts: []
+      - eval_time: 40m
+        alertname: RHACSEgressProxyReplicaCount
+        exp_alerts:
+          - exp_labels:
+              alertname: RHACSEgressProxyReplicaCount
+              namespace: rhacs-aaaabbbbccccddddeeee
+              deployment: egress-proxy
+              # not sure why the observed metrics have container=kube-rbac-proxy-main
```

Comment:
I think this is because the kube metrics are scraped by OpenShift platform monitoring over a secured endpoint, where the authn/z of `/metrics` is handled by kube-rbac-proxy.